### PR TITLE
Create a global user badge component with title

### LINKED ISSUE #4690: Add tooltips with full group name where abbreviations are used for display
It doesn't seem to be taking too much place while it used to be that way and it would be more clear I think.

### PR DESCRIPTION
**Pull request by TPGPL**
Resolves #4690

### CODE REVIEW

File: resources/lang/en/beatmap_discussions.php [hunk 1]
```diff
@@ -78,4 +78,11 @@
         'everyone' => 'Everyone',
         'label' => 'Filter by user',
     ],
+
+    'user_title' => [
+        'mapper' => 'Beatmap Creator',
+        'bng' => 'Beatmap Nominator',
+        'bng_limited' => 'Probationary Beatmap Nominator',
+        'nat' => 'Nomination Assessment Team Member',
```
Comment (VenixOSU) [inline resources/lang/en/beatmap_discussions.php]: not sure if "member" is needed

File: resources/assets/coffee/react/beatmap-discussions/post.coffee [hunk 2]
```diff
@@ -113,7 +113,9 @@ export class Post extends React.PureComponent
             div
               className: "#{bn}__user-badge"
               if userBadge?
-                div className: "user-group-badge user-group-badge--#{userBadge}"
+                div
+                  className: "user-group-badge user-group-badge--#{userBadge}"
+                  title: osu.trans "beatmap_discussions.user_title.#{userBadge}" if !@props.user.is_bot
```
Comment (notbakaneko) [inline resources/assets/coffee/react/beatmap-discussions/post.coffee]: why no title for bot?

File: resources/lang/en/beatmap_discussions.php [hunk 3]
```diff
@@ -78,4 +78,11 @@
         'everyone' => 'Everyone',
         'label' => 'Filter by user',
     ],
+
+    'user_title' => [
```
Comment (notbakaneko) [inline resources/lang/en/beatmap_discussions.php]: missing all the other groups
https://github.com/ppy/osu-web/blob/8e2da62880bda8a163265b410666bf06a11da0d8/resources/assets/coffee/osu_common.coffee#L257

File: resources/assets/coffee/react/beatmap-discussions/post.coffee (hunk 2 above)
Comment (TPGPL) [reply resources/assets/coffee/react/beatmap-discussions/post.coffee]: i thought its self explanatory already

Comment (notbakaneko) [reply resources/assets/coffee/react/beatmap-discussions/post.coffee]: If there's going to be a tooltip there, it should be for all of them

Comment (VenixOSU) [reply resources/assets/coffee/react/beatmap-discussions/post.coffee]: to be honest, i think not having a tooltip for bot is reasonable, it's a special case and also tooltip would need to say same as badge what's quite dumb

**Comment by TPGPL** [comment]
added a special bot tooltip which can give more context about it i think

**Comment by nekodex** [comment]
BTW these tooltips shouldn't be specific to beatmap modding - the badges are going to be used globally so the tooltips need to also be global.

Also the group names should NOT be translated, they should be using whatever names they've been given in the database.

**Comment by nekodex** [comment]
You're still using translations for the group name...

File: resources/lang/en/common.php [hunk 4]
```diff
@@ -108,18 +108,6 @@
         'target' => 'drop here to upload',
     ],
 
-    'group_name' => [
-        'bot' => 'Automated Message',
```
Comment (VenixOSU) [inline resources/lang/en/common.php]: I'm pretty sure that "automated message" won't really work globally, would be better to replace it with something more universal so it doesn't fit on modding discussions (and possibly some other message stuff) only.

File: resources/assets/coffee/react/_components/user-badge.coffee [hunk 5]
```diff
@@ -19,8 +19,19 @@
 import * as React from 'react'
 import { div } from 'react-dom-factories'
 bn = 'user-group-badge'
+group_names =
+  bot: 'Automated Message'
```
Comment (VenixOSU) [inline resources/assets/coffee/react/_components/user-badge.coffee]: I'm pretty sure that "automated message" won't really work globally, would be better to replace it with something more universal so it doesn't fit on modding discussions (and possibly some other message stuff) only.

Comment (peppy) [reply resources/assets/coffee/react/_components/user-badge.coffee]: also pretty sure this one will want to be localisable

Comment (TPGPL) [reply resources/assets/coffee/react/_components/user-badge.coffee]: so finally should those strings be localisable or not?

**Comment by TPGPL** [comment]
i don't think i will be able to do that as i don't know typescript 🤷‍♂

**Comment by notbakaneko** [comment]
This needs to be typescript

**Pull request closed without merging**